### PR TITLE
Update Middleware.php

### PR DESCRIPTION
**Pull request by spiffyjr**

### CODE REVIEW

File: src/Middleware.php [hunk 1]
```diff
@@ -91,12 +91,13 @@ public function __invoke(Request $request, Response $response, callable $out = n
      *
      * A handler can be any callable, or an object with a handle() method.
      *
-     * Handlers with arity >= 4 are considered error handlers, and will
+     * Handlers with parity >= 4 are considered error handlers, and will
```
Comment (weierophinney) [inline src/Middleware.php]: Actually, "arity" is the correct term here; see https://www.google.com/search?q=define%3Aarity

File: src/Middleware.php [hunk 2]
```diff
@@ -91,12 +91,13 @@ public function __invoke(Request $request, Response $response, callable $out = n
...
      * be executed when a handler calls $next with an argument or raises
      * an exception.
      *
      * @param string|callable|object $path Either a URI path prefix, or a handler
      * @param null|callable|object $handler A handler
+     * @return $this|void
```
Comment (weierophinney) [inline src/Middleware.php]: We can remove `void` as a return, and rename `$this` to `self` (per phpdoc guidelines).

**Comment by weierophinney** [comment]
Build problems are due to some syntax in the tests that only work with 5.5 and up -- but the other points need to be corrected. :)

**Comment by spiffyjr** [comment]
Done.

**Comment by weierophinney** [comment]
Awesome, thanks.

**Pull request merged**